editor/src/components/custom-code/code-file.spec.ts

```diff
@@ -407,3 +442,179 @@ describe('Creating require function', () => {
+import { jsx } from 'utopia-api'
+import { Card } from '/src/card.js'
+export var App = (props) => {
+  return <div data-uid='app-outer-div'>
+    <Card data-uid='card-instance' />
+  </div>
+}
+`,
+    ),
+    '/src/card.js': createCodeFile(
+      '/src/app.js',
+      `/** @jsx jsx */
+import * as React from 'react'
+import { jsx, Rectangle } from 'utopia-api'
+export var Card = (props) => {
+  return <div data-uid='card-outer-div'>
+    <div data-uid='card-inner-div' />
+    <Rectangle data-uid='card-inner-rectangle' /> 
+  </div>
+}
+`,
+    ),
+    '/utopia/unparsedstoryboard.js': createCodeFile(
+      StoryboardFilePath,
+      `/** @jsx jsx */
+import * as React from 'react'
+import { Scene, Storyboard, jsx } from 'utopia-api'
+import { App } from '/src/app.js'
+export var storyboard = (
+  <Storyboard data-uid='storyboard-entity'>
+    <Scene
+      data-uid='scene-entity'
+      component={App}
+      props={{}}
+      style={{ position: 'absolute', left: 0, top: 0, width: 375, height: 812 }}
+    />
+  </Storyboard>
+)
+`,
+    ),
+  }
+
+  projectContents = objectMap((projectFile: ProjectFile, fullPath: string) => {
+    if (isTextFile(projectFile) && fullPath !== '/utopia/unparsedstoryboard.js') {
+      const code = projectFile.fileContents.code
+      const parsedFile = parseCode(fullPath, code)
+      return textFile(textFileContents(code, parsedFile, RevisionsState.BothMatch), null, 1000)
+    } else {
+      return projectFile
+    }
+  }, projectContents)
+
+  return contentsToTree(projectContents)
+}
+
+function getTextFileByPath(projectContents: ProjectContentTreeRoot, path: string): TextFile {
+  const possibleResult = getContentsTreeFileFromString(projectContents, path)
+  if (isTextFile(possibleResult)) {
+    return possibleResult
+  } else {
+    throw new Error(`Unable to find a text file at path ${path}.`)
+  }
+}
+
+function staticInstancePathFromString(path: string): StaticInstancePath {
+  const fromStringResult = TP.fromString(path)
+  if (TP.isScenePath(fromStringResult)) {
+    throw new Error(`${path} represents a scene path.`)
+  } else {
+    return TP.dynamicPathToStaticPath(fromStringResult)
+  }
+}
+
+describe('normalisePathToUnderlyingTarget', () => {
+  const projectContents = defaultProjectContentsForNormalising()
+  it('jumps across multiple files to reach the actual target', () => {
+    const actualResult = normalisePathToUnderlyingTarget(
+      projectContents,
+      StoryboardFilePath,
+      staticInstancePathFromString(
+        'storyboard-entity/scene-entity:app-outer-div/card-instance:card-outer-div/card-inner-div',
+      ),
+    )
+    const expectedResult = normalisePathSuccess(
+      staticInstancePathFromString(':card-outer-div/card-inner-div'),
+      '/src/card.js',
+      getTextFileByPath(projectContents, '/src/card.js'),
+    )
+    expect(actualResult).toEqual(expectedResult)
+  })
+  it('returns the same path because there are no hops to take', () => {
```

Comment:
Please can you add a test case for covering no hops with an instance path that also has a scene path, so that we can ensure that's not a weird potential edge case.

Reply:
Fixed in 9bf80891.